File: dotnet/Vaas/src/Vaas/Messages/AuthenticationRequest.cs
using System.Text.Json.Serialization;

namespace Vaas.Messages
{
    public class AuthenticationRequest
    {
        [JsonPropertyName("kind")] public string Kind = "AuthRequest";

        [JsonPropertyName("token")] public string Token { get; }

        [JsonPropertyName("session_id")] public string? SessionId { get; }

        public AuthenticationRequest(string token, string? sessionId = null)
        {
            Token = token;
            SessionId = sessionId;
        }
    }
}    

File: golang/vaas/v2/pkg/messages/authentication_request.go
// Package messages provides structures for handling communication messages between the client and the VaaS server.
package messages

// AuthRequest represents an authentication request sent to the server.
type AuthRequest struct {
	Kind      string `json:"kind" default:"AuthRequest"`
	Token     string `json:"token"`
	SessionID string `json:"session_id" default:""`
}
